test/unit/DCAStrategies/dca-strategies-positions-handler.spec.ts

```diff
@@ -445,4 +456,211 @@ contract('DCAStrategiesPositionsHandler', () => {
+
+        const NEW_SHARE_A = { token: tokenA.address, share: BigNumber.from(20e2) };
+        const NEW_SHARE_B = { token: tokenB.address, share: BigNumber.from(35e2) };
+        const NEW_SHARE_D = { token: tokenD.address, share: BigNumber.from(20e2) }; // deposit
+        const NEW_SHARE_E = { token: tokenE.address, share: BigNumber.from(25e2) }; // deposit
+        newShares = sortTokens([NEW_SHARE_A, NEW_SHARE_B, NEW_SHARE_D, NEW_SHARE_E]);
+
+        hub.userPosition
+          .whenCalledWith(1)
+          .returns(createUserPosition(tokenF.address, amountOfSwaps, totalAmount, BigNumber.from(5), oldShares[0]));
+        hub.userPosition
+          .whenCalledWith(2)
+          .returns(createUserPosition(tokenF.address, amountOfSwaps, totalAmount, BigNumber.from(5), oldShares[1]));
+        hub.userPosition
+          .whenCalledWith(3)
+          .returns(createUserPosition(tokenF.address, amountOfSwaps, totalAmount, BigNumber.from(5), oldShares[2]));
+
+        hub.terminate
+          .whenCalledWith(1, DCAStrategiesPositionsHandlerMock.address, user.address)
+          .returns([totalAmount.mul(oldShares[0].share).div(100e2), 0]);
+        hub.terminate
+          .whenCalledWith(2, DCAStrategiesPositionsHandlerMock.address, user.address)
+          .returns([totalAmount.mul(oldShares[1].share).div(100e2), 0]);
+        hub.terminate
+          .whenCalledWith(3, DCAStrategiesPositionsHandlerMock.address, user.address)
+          .returns([totalAmount.mul(oldShares[2].share).div(100e2), 0]);
+
+        hub.reducePosition.returns(true);
+        hub.increasePosition.returns(true);
+
+        hub['deposit(address,address,uint256,uint32,uint32,address,(address,uint8[])[])'].returnsAtCall(0, 4); // at call 0 return id 4
+        hub['deposit(address,address,uint256,uint32,uint32,address,(address,uint8[])[])'].returnsAtCall(1, 5); // at call 1 return id 5
+
+        tokenF.transfer.returns(true);
+        tokenF.transferFrom.returns(true);
+
+        await DCAStrategiesPositionsHandlerMock.setPermissions(true);
+        await DCAStrategiesPositionsHandlerMock.setTokenShares(3, newShares);
+        await DCAStrategiesPositionsHandlerMock.setUserPositions(1, {
+          strategyId: 1,
+          strategyVersion: 1,
+          hub: hub.address,
+          positions: positions,
+        });
+      });
+      when('is called (trying to reduce position)', () => {
+        given(async () => {
+          tx = await callSync(totalAmount.sub(delta), newAmountOfSwaps);
+
+          newPositions = (await DCAStrategiesPositionsHandlerMock.userPosition(1)).positions;
+        });
+        then('transfer() is called correctly', async () => {
+          expect(tokenF.transfer).to.have.been.calledWith(user.address, delta);
+        });
+        then('increase, reduce, deposit or terminate is called correctly', async () => {
+          checkCallsToHub();
+          expect(hub.reducePosition).to.have.been.calledTwice;
```

Comment:
Could we check that each reduce was called with the correct amount? I know it's a pain, but I think it's very important to understand that we are doing everything correctly